### DISCUSSION TITLE
How to update state into particle options in real-time while preventing re-rendering?

### DISCUSSION THREAD

**Discussion by gyeongjin-kim** [matteobruni/tsparticles#5416] [Q&A] [answered]
Hello, I am developing with React and using @tsparticles/react 3.0.0. I am trying to pass a state that updates in real-time as a prop to the Particle component so that it can dynamically change the options, such as the number value or animation speed.

The issue I'm facing is that when I pass the state as a prop, the Particle component re-renders whenever the state updates, which results in the particle effect being completely restarted. I would like the particle effect to smoothly update its options without restarting the animation each time the state changes.

Are there any recommended approaches or best practices to achieve this behavior?

Any help would be greatly appreciated!

**Comment by matteobruni** [comment, marked as answer]
Duplicate of https://github.com/tsparticles/react/issues/122
